common/src/test/kotlin/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketCoreImplTest.kt

```diff
@@ -0,0 +1,690 @@
+        webSocketListener = webSocketCore
+        every {
+            mockOkHttpClient.newWebSocket(
+                any(),
+                webSocketListener,
+            )
+        } returns mockConnection
+    }
+
+    private fun prepareAuthenticationAnswer(
+        haVersion: String = "2025.4.1",
+        successfulAuth: Boolean = true,
+    ) {
+        // Simulate that the server receive the auth message and answer that it is accepted
+        every {
+            mockConnection.send(
+                match<String> { message ->
+                    message.contains(""""type":"auth"""") &&
+                        message.contains(""""access_token":"mock_access_token"""")
+                },
+            )
+        } answers {
+            assertSame(WebSocketState.AUTHENTICATING, webSocketCore.getConnectionState())
+            webSocketListener.onMessage(
+                mockConnection,
+                """{"id":1,"type":"${if (successfulAuth) "auth_ok" else "auth_invalid"}","ha_version":"$haVersion"}""",
+            )
+            true
+        }
+    }
+
+    private fun mockResultSuccessForId(id: Long) {
+        every { mockConnection.send(match<String> { it.contains(""""id":$id""") }) } answers {
+            webSocketListener.onMessage(
+                mockConnection,
+                """{"id":$id,"type":"result","success":true,"result":{}}""",
+            )
+            true
+        }
+    }
+
+    private fun closeConnection() {
+        webSocketListener.onClosed(mockConnection, 1000, "test")
+    }
+
+    /*
+connect()
+     */
+
+    @ParameterizedTest
+    @ValueSource(strings = ["", "htt://io.ha", "ws://io.ha", "wss://io.ha"])
+    fun `Given invalid url When connect is invoked Then it returns false and connection state is null`(
+        url: String,
+    ) = runTest {
+        setupServer(url)
+
+        val result = webSocketCore.connect()
+
+        assertFalse(result)
+        assertNull(webSocketCore.getConnectionState())
+    }
+
+    @Test
+    fun `Given no server When connect is invoked Then it returns false and connection state is null`() =
+        runTest {
+            val serverManager = mockk<ServerManager>(relaxed = true)
+            every { serverManager.getServer(any<Int>()) } returns null
+
+            val webSocketCore = WebSocketCoreImpl(
+                okHttpClient = mockk(),
+                serverManager = serverManager,
+                serverId = 1,
+            )
+
+            val result = webSocketCore.connect()
+
+            assertFalse(result)
+            assertNull(webSocketCore.getConnectionState())
+        }
+
+    @Test
+    fun `Given failure to connect after timeout When connect is invoked Then it returns false and connection state is AUTHENTICATING`() =
+        runTest {
+            setupServer()
+
+            val result = webSocketCore.connect()
+
+            assertFalse(result)
+            // TODO do we still want to be authenticating at this stage?
```

Comment:
Good point, I think the answer is no. That comment should get another place though.

Reply:
I fixed it and remove the TODO

Reply:
I will handle that in the follow up PR actually since it's a bit more complicated.